libraries/core_libs/network/src/tarcap/shared_states/syncing_state.cpp

```diff
@@ -10,6 +10,10 @@ SyncingState::SyncingState(uint16_t deep_syncing_threshold)
 
 void SyncingState::set_peer(std::shared_ptr<TaraxaPeer> &&peer) {
   std::unique_lock lock(peer_mutex_);
+  // we set peer to null if dag and pbft syncing are not enabled
+  if (!peer && (pbft_syncing_ || dag_syncing_)) {
```

Comment:
I think it is the tough part of code to read. Could we explicit set it to null at least?

Reply:
I guess it is just easier to read then that one. But I am not sure if there could be a situation that value are other then null value here. 
```
if (!pbft_syncing_ && !dag_syncing) {
  peer_ = nullptr;
}
```